k2/csrc/fsa_equivalent_test.cc

```diff
@@ -0,0 +1,61 @@
+// k2/csrc/fsa_equivalent_test.cc
+
+// Copyright (c)  2020  Haowen Qiu
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/csrc/fsa_equivalent.h"
+
+#include <utility>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "k2/csrc/fsa.h"
+
+namespace k2 {
+
+TEST(Properties, RandomPathFail) {
+  {
+    Fsa fsa;
+    Fsa path;
+    bool status = RandomPath(fsa, &path);
+    EXPECT_FALSE(status);
+  }
+  // TODO(haowen): add tests for non-connected fsa
+}
+
+TEST(Properties, RandomPathSuccess) {
+  {
+    Fsa fsa;
+    std::vector<Arc> arcs = {
+        {0, 1, 1}, {0, 2, 2}, {1, 2, 3}, {2, 3, 4},
+        {2, 4, 5}, {3, 4, 7}, {4, 5, 9},
+    };
+    std::vector<int32_t> arc_indexes = {0, 2, 3, 5, 6, 7};
+    fsa.arc_indexes = std::move(arc_indexes);
+    fsa.arcs = std::move(arcs);
+    Fsa path;
+
+    {
+      bool status = RandomPath(fsa, &path);
+      EXPECT_TRUE(status);
+    }
+
+    {
+      for (size_t i = 0; i < 20; ++i) {
+        std::vector<int32_t> state_map;
```

Comment:
`state_map` can be moved out of the `for` loop.

Reply:
Sure